diff --git a/include/linux/lsm_hooks.h b/include/linux/lsm_hooks.h
index 000000000000..111111111111 100644
--- a/include/linux/lsm_hooks.h
+++ b/include/linux/lsm_hooks.h
@@ -192,4 +192,10 @@ static inline struct xattr *lsm_get_xattr_slot(struct xattr *xattrs,
 	return &xattrs[(*xattr_count)++];
 }
 
+#ifdef CONFIG_SECURITY_HONEYBEST
+extern void __init honeybest_add_hooks(void);
+#else
+static inline void __init honeybest_add_hooks(void) { }
+#endif
+
 #endif /* ! __LINUX_LSM_HOOKS_H */
diff --git a/security/Kconfig b/security/Kconfig
index 000000000000..111111111111 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -225,6 +225,7 @@ source "security/tomoyo/Kconfig"
 source "security/apparmor/Kconfig"
 source "security/loadpin/Kconfig"
 source "security/yama/Kconfig"
+source "security/honeybest/Kconfig"
 source "security/safesetid/Kconfig"
 source "security/lockdown/Kconfig"
 source "security/landlock/Kconfig"
diff --git a/security/Makefile b/security/Makefile
index 000000000000..111111111111 100644
--- a/security/Makefile
+++ b/security/Makefile
@@ -20,6 +20,7 @@ obj-$(CONFIG_SECURITY_TOMOYO)		+= tomoyo/
 obj-$(CONFIG_SECURITY_APPARMOR)		+= apparmor/
 obj-$(CONFIG_SECURITY_YAMA)		+= yama/
+obj-$(CONFIG_SECURITY_HONEYBEST)	+= honeybest/
 obj-$(CONFIG_SECURITY_LOADPIN)		+= loadpin/
 obj-$(CONFIG_SECURITY_SAFESETID)       += safesetid/
 obj-$(CONFIG_SECURITY_LOCKDOWN_LSM)	+= lockdown/
diff --git a/security/security.c b/security/security.c
index 000000000000..111111111111 100644
--- a/security/security.c
+++ b/security/security.c
@@ -20,6 +20,9 @@
 #include <linux/kernel_read_file.h>
 #include <linux/lsm_hooks.h>
+#ifdef CONFIG_SECURITY_HONEYBEST
+extern void __init honeybest_add_hooks(void);
+#endif
 #include <linux/mman.h>
 #include <linux/mount.h>
 #include <linux/personality.h>
@@ -543,6 +546,9 @@ int __init security_init(void)
 	/* Load LSMs in specified order. */
 	ordered_lsm_init();
 
+#ifdef CONFIG_SECURITY_HONEYBEST
+	honeybest_add_hooks();
+#endif
+
 	return 0;
 }
